packages/syft/tests/syft/users/user_service_test.py

```diff
@@ -604,3 +625,37 @@ def mock_get_by_email(credentials: SyftVerifyKey, email: str) -> Err:
     response = user_service.exchange_credentials(unauthed_context)
     assert isinstance(response, SyftError)
     assert response.message == expected_error_msg
+
+
+def test_userservice_toggle_registration(
+    faker, guest_domain_client, root_domain_client
+) -> None:
+    email1 = faker.email()
+    email2 = faker.email()
+    response_1 = root_domain_client.register(
+        email=email1, password="joker123", name="Joker"
+    )
+    assert isinstance(response_1, SyftSuccess)
+    # by default, the guest client can't register new user
+    response_2 = guest_domain_client.register(
+        email=email2, password="harley123", name="Harley"
+    )
+    assert isinstance(response_2, SyftError)
+
+    assert any([user.email == email1 for user in root_domain_client.users])
+
+    # only after the root client enable other users to signup, they can
```

Comment:
a DO client would still be able to register users right? Can we add a test for that?

Reply:
Yes, they can. Will add a test for it.